src/meta/service/src/ddl_service.rs

```diff
@@ -966,6 +967,46 @@ impl DdlService for DdlServiceImpl {
                 .await?;
 
             for table in tables {
+                // Since we only support `ADD` and `DROP` column, we check whether the new columns and the original columns
+                // is a subset of the other.
+                let original_column_names: HashSet<String> = HashSet::from_iter(
+                    table
+                        .columns
+                        .iter()
+                        .map(|col| ColumnCatalog::from(col.clone()).column_desc.name),
+                );
+                let new_column_names: HashSet<String> = HashSet::from_iter(
+                    table_change
+                        .columns
+                        .iter()
+                        .map(|col| ColumnCatalog::from(col.clone()).column_desc.name),
+                );
+                if !(original_column_names.is_subset(&new_column_names)
+                    || original_column_names.is_superset(&new_column_names))
```

Comment:
I just realize that comparing the name sets is not sufficient, as users may alter the data type of a column, or even set a new default value. Those functionalities are not supported by us.

Reply:
Yes. So there is one more check in below.
```
                // skip the schema change if there is no change to original columns
                if original_column_names == new_column_names {
                    continue;
                }
```

Reply:
What if 

1. alter column type, we skip since there's no change on the names
2. then, add column, we do schema change. Will we also accidentally apply the changes happened in step 1?

Reply:
> Will we also accidentally apply the changes happened in step 1?

Yes, that would be a problem. Since we apply the latest version column schema from upstream.

Reply:
Thanks for point out the issue, I submit the patch in 
https://github.com/risingwavelabs/risingwave/pull/18322/commits/f112e6d619e085aa1caa4d07bd2c0789af6128cd. (#18322)